### PR TITLE
better specify coordinate meanings, fix related bugs, bounds checks

### PR DESCRIPTION
**Pull request by aaronjanse**
Fixes many one-off errors. Also adds bounds checks to make the fuzzer happy.

### CODE REVIEW

File: vterm/vterm.go [hunk 1]
```diff
@@ -21,7 +23,10 @@ VTerm acts as a virtual terminal emulator between a shell and the host terminal
 It both transforms an inbound stream of bytes into Char's and provides the option of dumping all the Char's that need to be rendered to display the currently visible terminal window from scratch.
 */
 type VTerm struct {
-	x, y, w, h int
+	// starting at (0, 0) in top left
+	x, y int
+	// x < w && y < h
+	w, h int
```
Comment (aaronjanse) [inline vterm/vterm.go:29]: Note now-explicit coordinate meanings

File: vterm/vterm.go [hunk 2]
```diff
@@ -11,7 +11,9 @@ import (
 
 // ScrollingRegion holds the state for an ANSI scrolling region
 type ScrollingRegion struct {
-	top    int
+	// cursor within when cursor.Y >= top
+	top int
+	// cursor within when cursor.Y < bottom
```
Comment (aaronjanse) [inline vterm/vterm.go:16]: Note now-explicit coordinate meanings

File: render/render.go [hunk 3]
```diff
@@ -111,12 +111,14 @@ func (r *Renderer) ListenToQueue() {
 		for {
 			fullyWritten := true
 			var diff strings.Builder
+		outer:
 			for y := 0; y <= r.h; y++ {
 				for x := 0; x < r.w; x++ {
 					// some terminals truncate long stdout
+					// 4000 to allow some extra chars to be added before crossing 4096
```
Comment (PotatoParser) [inline render/render.go:118]: Good point!

**Review by PotatoParser — APPROVED** [review]
Looking very very nice! Thanks for the super helpful comments!

**Pull request merged**